akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2.scala

```diff
@@ -213,19 +226,24 @@ private[http] final class Http2Ext(private val config: Config)(implicit val syst
   }
 
   def outgoingConnectionPriorKnowledge(host: String, port: Int, clientConnectionSettings: ClientConnectionSettings, log: LoggingAdapter): Flow[HttpRequest, HttpResponse, Future[OutgoingConnection]] = {
-    val stack = Http2Blueprint.clientStack(clientConnectionSettings, log) atop
+    val stack = Http2Blueprint.clientStack(clientConnectionSettings, log, telemetry).addAttributes(prepareClientAttributes(host, port)) atop
       Http2Blueprint.unwrapTls atop
       LogByteStringTools.logTLSBidiBySetting("client-plain-text", clientConnectionSettings.logUnencryptedNetworkBytes) atop
       TLSPlacebo()
 
     stack.joinMat(clientConnectionSettings.transport.connectTo(host, port, clientConnectionSettings)(system.classicSystem))(Keep.right)
   }
 
+  private def prepareClientAttributes(serverHost: String, port: Int): Attributes =
+    if (telemetry == NoOpTelemetry) Attributes.none
+    else TelemetryAttributes.prepareClientFlowAttributes(serverHost, port)
+
 }
 
 /** INTERNAL API */
 @InternalApi
 private[http] object Http2 extends ExtensionId[Http2Ext] with ExtensionIdProvider {
+  @InternalStableApi
```

Comment:
Why?

Reply:
We use the streamId key in telemetry, even though it is internal, to correlate client response with the request it is for.

Reply:
Sorry, I'm wrong, we stopped doing that and are using our own attribute, I'll remove this